File: src/Thunk/Thunk.cpu.Tests/System/Collections/ComparerTests.cpp
using namespace Xunit;
using namespace System;

/*
Declare_Collections_IComparer(System_String, string)
Declare_Collections_Comparer(System_String, string)

//#region Sample

public class(SampleComparer,Collections_Comparer(System_String),,
);
Declare_DefaultCtor(public,SampleComparer)

private_ int_ Compare(Collections_Comparer(System_String) *t, string x, string y)
{
return 0;
}
Implement_Type(SampleComparer,Collections_Comparer(System_String),,
System_ObjectVector *vtable = t->Collections_Comparer(System_String).__vtable;
vtable[_Collections_Comparer_System_StringVIndex_Compare] = (System_ObjectVector)&space(SampleComparer,Compare);
)
Implement_DefaultCtor(SampleComparer,Collections_Comparer(System_String))

//#endregion Sample
*/

namespace System { namespace Collections {
	/// <summary>
	/// ComparerTests
	/// </summary>
	public ref class ComparerTests
	{
	public:
		/*
		[Fact]
		void Constructor_Valid_NotNull()
		{
		SampleComparer *comparer = new(,SampleComparer);
		Assert::NotNull(comparer);
		}

		[Fact]
		void Compare_Valid_EqualsZero()
		{
		SampleComparer *comparer = new(,SampleComparer);
		Assert::Equal(0, vcallT(Collections_Comparer,System_String,comparer,Compare, L"test", L"test"));
		}

		[Fact]
		void getDefault_Valid_NotNull()
		{
		Assert::NotNull(getsT(Collections_Comparer,System_String,Default));
		}
		*/
	};
}}

File: src/Thunk/Thunk.Tests/System/Collections/ComparerTests.cs
using Xunit;
using System.Collections.Generic;
namespace System.Collections
{
    #region Fakes

    /// <summary>
    /// FakeComparer
    /// </summary>
    internal class FakeComparer : Comparer<string>
    {
        public override int Compare(string x, string y) { return 0; }
    }

    #endregion

    /// <summary>
    /// ComparerTests
    /// </summary>
    public class ComparerTests
    {
        public void Constructor_Valid_NotNull()
        {
            var comparer = new FakeComparer();
            Assert.NotNull(comparer);
        }

        [Fact]
        public void Compare_Valid_EqualsZero()
        {
            var comparer = new FakeComparer();
            Assert.Equal(0, comparer.Compare("test", "test"));
        }

        [Fact]
        public void getDefault_Valid_NotNull()
        {
            Assert.NotNull(Comparer<string>.Default);
        }

        //[Fact]
        //public void Equals_Valid_EqualsTrue()
        //{
        //    var comparer = new FakeComparer();
        //    Assert.True(comparer.Equals(comparer));
        //}

        //[Fact]
        //public void GetHashCode_Valid_NotEqualsZero()
        //{
        //    var comparer = new FakeComparer();
        //    Assert.NotEqual(0, comparer.GetHashCode());
        //}

        //[Fact]
        //public void GetType_Valid_SameAsSampleComparerType()
        //{
        //    var comparer = new FakeComparer();
        //    Assert.Same(comparer.GetType(), typeof(SampleComparer));
        //}

        //[Fact]
        //public void ToString_Valid_EqualsSampleComparerTypeAsText()
        //{
        //    var comparer = new FakeComparer();
        //    Assert.Equal("ThunkC_.Collections_.SampleComparer", comparer.ToString());
        //}
    }
}
